ducktape/cluster/cluster.py

```diff
@@ -38,21 +45,39 @@ def __len__(self):
         """Size of this cluster object. I.e. number of 'nodes' in the cluster."""
         raise NotImplementedError()
 
-    def request(self, nslots):
-        """Request the specified number of slots, which will be reserved until they are freed by the caller."""
+    def alloc_subcluster(self, num_nodes):
```

Comment:
Hmm, this sounds like a convenience around `alloc`. Why wouldn't there be a generic implementation of it?